feat: Jira/CS-15 - Add services-card component and setup tooling

hbabb opened the pull request:
**Summary**
This PR adds the ```services``` component and improves project tooling and consistency.

**Changes**
- Refactored ```layout.tsx``` to remove unused styles and fonts
- Integrated ```services.tsx``` into ```page.tsx``` (SPA style project)
- Created ```data/services.ts``` to manage company services data
- Developed ```lib/hooks/useScreenSize.ts``` for mobile detection
- Imported ShadCN/UI's *Card* and *Accordion* components
- Configured *Prettier*, updated *ES Lint*, and enabled Tailwindcss sorting
- Added VSCode settings for cspell and formatting/linting consistency 
- Installed Autoprefixer and updated PostCSS Configuration
- Styled ```services.tsx``` component according to Figma design
- Fixed markdown linting issues in CONTRIBUTING.md
- Documented code with comments for clarity

**Impact**
- Enhances project structure and maintainability
- Improves styling and layout consistency
- Standardizes development environment across team members

**Testing**
- Manually verified component integration
- Ensured Prettier and ESLint apply formatting correctly
- Validated ```isMobile``` logic

**Linked Issue**
Resolves Jira issue CS-15

Commit:
feat: Jira/CS-15 - Add services-card component and setup tooling

- Removed unused styles and fonts from layout.tsx
- Integrated services-card component into main page.tsx (SPA)
- Created `data/services.ts` for company services
- Built `lib/hooks/useScreenSize.ts` to manage `isMobile` state
- Added Card and Accordion components from ShadCN/UI
- Configured Prettier and updated ESLint to align with it
- Enabled Tailwind CSS sorting in Prettier
- Added VSCode settings for cspell and formatting consistency
- Installed Autoprefixer and updated PostCSS config
- Styled services-card per Figma design
- Fixed markdown lint issues in `CONTRIBUTING.md`
- Added inline code documentation

Completes Jira issue **CS-15**

AnnieMcMahon commented on app/services/services.tsx:
I would suggest capitalizing the file name, "Services.tsx", to clearly show that this is a component. 

I was talking with Michael (MT.M) yesterday about folder structure. He added a "_components" folder in his code so we can keep all the components together instead of in separate folders. Once his PR is approved, I will move my component in that folder. There is no need to create separate folders because they are not pages.

hartmandy replied:
ShadCN also makes it's own components folder, I wouldn't keep Shad and non Shad in the same folder.

AnnieMcMahon commented on app/services/services.tsx:
```diff
@@ -0,0 +1,99 @@
+ * and a grid of cards for desktop devices.
+ *
+ * @returns The Services component.
+ */
```
I don't think comments are needed here. Your code is pretty clear already. I would actually remove all the comments and let the code speak for itself.

AnnieMcMahon commented on app/services/services.tsx:
```diff
@@ -0,0 +1,99 @@
+      </div>
+    </section>
+  );
+}
```
Your component looks really nice! Your font is bigger than the one I use. We'll have to decide which one we want and use it consistently throughout the project.

hartmandy replied:
Agreed. It's beautifully done Heath, good job! I would just match fonts. I have no preference on which one, I'll leave that up to y'all.

hbabb replied:
@AnnieMcMahon Do you have a preference on the font? I just made mine match the figma design, but it is easy to change if you want to match to the font size you set. I didn't go into your file to see what size you chose, so just let me know which way you want to go.

AnnieMcMahon commented on data/services.ts:
```diff
@@ -0,0 +1,40 @@
+// An array of objects representing the services offered by Classics Revisited
```
I would remove all the comments here too.

AnnieMcMahon commented on lib/hooks/useScreenSize.ts:
```diff
@@ -0,0 +1,44 @@
+  }, []);
+
+  return { isMobile };
+}
```
Waaay too many comments! I would remove all of them. This is a great hook that I might steal for my other projects.

AnnieMcMahon replied:
I just noticed that Michael has a similar hook in his PR called "use-mobile.tsx". We may need to decide which one to keep and eliminate one of them at some point.

hartmandy replied:
I said this on Michael's PR too, but we're using Tailwind so you can do sm, md, lg, and xl in your className to do things like this. md:hidden, or sm:w-full, etc. is how you typically handle screen sizes changes. https://tailwindcss.com/docs/responsive-design

Mountain-Michaelle replied:
Yes but the hook is from shadcn component extension

hartmandy replied:
I've never noticed that file - ignore me then!

hbabb replied:
@hartmandy I agree with @Mountain-Michaelle that the hook is useful here, and I know we can do a lot with responsive design in tailwind, but how would we change between the cards and accordion or change the header to a sidecard for the navigation easily? I know we can use the hidden, but isn't the hook a better option for these task, making the code more readable and maintainable?

Thanks for all the help with this project and helping me understand things.

AnnieMcMahon reviewed (commented):
I like that you refactored the code and added some tools for consistency and simplicity. Your code is clear and easy to read, and your component looks great, both on large screens and on mobile. Great job!

Commit:
Merge remote-tracking branch 'origin/main' into heath/services-card

Commit:
chore: moved services to _components directory

Commit:
chore: refactored services.tsx useScreenSize.ts and services.ts

Removed all comments from the files. I kept a couple of comments to help with location in code for Services.tsx.

hbabb commented:
Closing this request. I have pushed the current revisions and will open a new request.
